pkg/addonmanager/addonclients/fluxaddonclient.go

```diff
@@ -95,9 +96,13 @@ func NewGitOptions(ctx context.Context, cluster *v1alpha1.Cluster, fluxConfig *v
 }
 
 func NewFluxAddonClient(flux Flux, gitOpts *GitOptions) *FluxAddonClient {
+	if gitOpts == nil {
```

Comment:
this is an intermediate step to replicate the existing behavior which will be removed once we completely remove the gitOpts and pass in individual dependencies; a different method for determining wether GitOps install should be skipped will be put in place.

Reply:
by 'existing behavior', which check if the 'gitOpts' is nil to determine wether ot install flux or not; we're now just checking wether FluxAddonClient itself is nil